### PR TITLE
WF1 Morecast2 Auth

### PR DESCRIPTION
**Pull request by conbrad**
Authenticates forecasters with WF1.

# Test Links:
[Landing Page](https://wps-pr-2854.apps.silver.devops.gov.bc.ca/)
[MoreCast 2.0](https://wps-pr-2854.apps.silver.devops.gov.bc.ca/morecast-2)
[Percentile Calculator](https://wps-pr-2854.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-2854.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2854.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2854.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2854.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-2854.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-2854.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

**Comment by dgboss** [comment]
Temporarily closing to free up resources.

**Review by andrea-williams — APPROVED** [review]
Code LGTM. Haven't tested it locally to ensure I can auth with WebADE.

I know there were some issues you had parsing the token in the URL - do you think it's worthwhile documenting the ouija spirits you had to summon to get that to work? (Or extending that, documenting somewhere that WebADE is now integrated into our app?

**Review by dgboss — COMMENTED** [review]
Also, do you think it would be possible to do one last redirect where we do a final page load with just the app URL so we're not showing the access token in URL?

**Review by dgboss — APPROVED** [review]

File: web/src/features/auth/components/MoreCast2AuthWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+interface Props {
+  children: React.ReactElement
+}
+
+const MoreCast2AuthWrapper = ({ children }: Props) => {
+  const dispatch: AppDispatch = useDispatch()
+
+  const [renderChildren, setRenderChildren] = useState(false)
+  const { isAuthenticated, roles } = useSelector(selectAuthentication)
```
Comment (dgboss) [inline web/src/features/auth/components/MoreCast2AuthWrapper.tsx:18]: I don't think we need `isAuthenticated` as this wrapper component won't even be rendered if we're not authenticated. We could look at it and remove in another PR.

Comment (conbrad) [reply web/src/features/auth/components/MoreCast2AuthWrapper.tsx]: Good call, removed. in https://github.com/bcgov/wps/pull/2854/commits/130a77fb077cc13f4c90926b4ddce7c7f2f8dd99

**Pull request merged**